services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -964,7 +964,10 @@ private void checkClientSession(boolean checkAction) {
         if (checkAction) {
             String action = AuthenticationSessionModel.Action.AUTHENTICATE.name();
             if (!code.isValidAction(action)) {
-                throw new AuthenticationFlowException(AuthenticationFlowError.INVALID_CLIENT_SESSION);
+                throw new AuthenticationFlowException("Action " + action +
```

Comment:
I think we should log the message instead, using the `debug` level perhaps.

Also, it seems the message is not really printing any code but the action.